diff --git a/properties/nm-openvpn-editor.c b/properties/nm-openvpn-editor.c
index 4c3b7a0..857c5e3 100644
--- a/properties/nm-openvpn-editor.c
+++ b/properties/nm-openvpn-editor.c
@@ -1064,7 +1064,7 @@ populate_tls_remote_mode_entry_combo (GtkEditable* entry, GtkComboBox *box,
 
 	gtk_list_store_append (store, &iter);
 	gtk_list_store_set (store, &iter,
-	                    TLS_REMOTE_MODE_COL_NAME, _("Don’t verify certificate identification"),
+	                    TLS_REMOTE_MODE_COL_NAME, _("Don't verify certificate identification"),
 	                    TLS_REMOTE_MODE_COL_VALUE, TLS_REMOTE_MODE_NONE,
 	                    -1);
 
@@ -1970,6 +1970,20 @@ advanced_dialog_new (GHashTable *hash, const char *contype)
 	                  "clicked", G_CALLBACK (gtk_widget_show),
 	                  gtk_builder_get_object (builder, "sk_key_chooser"));
 
+	widget = GTK_WIDGET (gtk_builder_get_object (builder, "scramble_type_combo"));
+	value = g_hash_table_lookup (hash, NM_OPENVPN_KEY_SCRAMBLE_TYPE);
+	if (value && *value)
+		gtk_combo_box_set_active_id(GTK_COMBO_BOX(widget), value);
+	else
+		gtk_combo_box_set_active(GTK_COMBO_BOX(widget), 0);
+
+	widget = GTK_WIDGET (gtk_builder_get_object (builder, "scramble_key_entry"));
+	value = g_hash_table_lookup (hash, NM_OPENVPN_KEY_SCRAMBLE_KEY);
+	if (value && *value)
+		gtk_editable_set_text(GTK_EDITABLE(widget), value);
+	else
+		gtk_editable_set_text(GTK_EDITABLE(widget), "");
+
 	return dialog;
 }
 
@@ -2417,6 +2431,20 @@ advanced_dialog_new_hash_from_dialog (GtkWidget *dialog)
 	if (gtk_check_button_get_active (GTK_CHECK_BUTTON (widget)))
 		g_hash_table_insert (hash, NM_OPENVPN_KEY_PUSH_PEER_INFO, g_strdup ("yes"));
 
+	widget = GTK_WIDGET (gtk_builder_get_object (builder, "scramble_type_combo"));
+	value = g_hash_table_lookup (hash, NM_OPENVPN_KEY_SCRAMBLE_TYPE);
+	if (value && *value)
+		gtk_combo_box_set_active_id(GTK_COMBO_BOX(widget), value);
+	else
+		gtk_combo_box_set_active(GTK_COMBO_BOX(widget), 0);
+
+	widget = GTK_WIDGET (gtk_builder_get_object (builder, "scramble_key_entry"));
+	value = g_hash_table_lookup (hash, NM_OPENVPN_KEY_SCRAMBLE_KEY);
+	if (value && *value)
+		gtk_editable_set_text(GTK_EDITABLE(widget), value);
+	else
+		gtk_editable_set_text(GTK_EDITABLE(widget), "");
+
 	return hash;
 }
 
