diff --git a/properties/import-export.c b/properties/import-export.c
index 51049de..f2346cd 100644
--- a/properties/import-export.c
+++ b/properties/import-export.c
@@ -1422,6 +1422,21 @@ do_import (const char *path, const char *contents, gsize contents_len, GError **
 			continue;
 		}
 
+		if (NM_IN_STRSET (params[0], "scramble")) {
+			if (!args_params_check_nargs_minmax (params, 1, 2, &line_error))
+				goto handle_line_error;
+			if (!args_params_check_arg_utf8 (params, 1, NULL, &line_error))
+				goto handle_line_error;
+			setting_vpn_add_data_item (s_vpn, NM_OPENVPN_KEY_SCRAMBLE_ENABLE, "yes");
+			setting_vpn_add_data_item (s_vpn, NM_OPENVPN_KEY_SCRAMBLE_TYPE, params[1]);
+			if (params[2]) {
+				if (!args_params_check_arg_utf8 (params, 2, NULL, &line_error))
+					goto handle_line_error;
+				setting_vpn_add_data_item (s_vpn, NM_OPENVPN_KEY_SCRAMBLE_KEY, params[2]);
+			}
+			continue;
+		}
+
 		if (NM_IN_STRSET (params[0], NMV_OVPN_TAG_IFCONFIG)) {
 			if (!args_params_check_nargs_n (params, 2, &line_error))
 				goto handle_line_error;
